### PR TITLE
TK-1 and TP-1 hardware production tests

### PR DESCRIPTION
**Pull request by cibomahto**

### CODE REVIEW

File: hw/boards/mta1-usb-v1/test/production_test.py [hunk 1]
```diff
@@ -83,8 +83,19 @@ def flash_validate_id():
             vals_b = line.split(b' 0x')[1:]
             flash_id = int(b''.join(vals_b),16)
             print(line, hex(flash_id))
-            if (flash_id == 0) or (flash_id == 0xFFFFFFFF):
+ 
+            # Note: Flash IDs reported by iceprog are slightly wrong
+            flash_types = {
+                0xb40140b40140b40140b40140b4014: 'XT25F08BDFIGT-S (MTA1-USB-V1)',
```
Comment (quite) [inline hw/boards/mta1-usb-v1/test/production_test.py]: @cibomahto is this correct? I see `0x0b40…`

Comment (cibomahto) [reply hw/boards/mta1-usb-v1/test/production_test.py]: Which hardware are you working with?

Comment (quite) [reply hw/boards/mta1-usb-v1/test/production_test.py]: Sorry, I somehow read the key of that dict as a string or something :D

Comment (cibomahto) [reply hw/boards/mta1-usb-v1/test/production_test.py]: Probably it should be comparing byte streams instead of integers!

**Pull request closed without merging**